cmd/decoder.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/spf13/cobra"
+)
+
+var (
+	keyFormat string
+	keyValue  string
+)
+
+// decoderCmd represents the key-decoder command
+var decoderCmd = &cobra.Command{
+	Use:   "decoder -f <key-format> -k <key>",
+	Short: "decode tabel_row/table_index/value format key to readable format",
+	Long:  "decode tabel_row/table_index/value format key to readable format",
+	RunE:  decodeKeyFunc,
+}
+
+var indexTypeToString = map[byte]string{
```

Comment:
If the type changed, we need to update it. Is it possible to avoid it?

Reply:
How about putting this map into TiDB? @jackysp